backend/amd64/proc.ml

```diff
@@ -133,18 +144,28 @@ let hard_float_reg =
   for i = 0 to 15 do v.(i) <- Reg.at_location Float (Reg (100 + i)) done;
   v
 
+let hard_vec128_reg =
+  let v = Array.make 16 Reg.dummy in
+  for i = 0 to 15 do v.(i) <- Reg.at_location Vec128 (Reg (100 + i)) done;
+  v
+
 let all_phys_regs =
-  Array.append hard_int_reg hard_float_reg
+  Array.concat [hard_int_reg; hard_float_reg; hard_vec128_reg]
```

Comment:
That does not feel right; the same register will 
appear in `hard_float_reg` and `hard_vec128_reg`
but the registers in `hard_int_reg` can be used
with different types. I am not positive this is
creating a problem, but the lack of homogeneity
gives me pause.

Reply:
Added a flag guarding the generatin of Vec128 registers, and validated that we get the exact same assembly output for flambda-backend:main / this pr with the flag enabled and disabled. I'm pretty confident this is fine now; we could even remove the flag again.